Shell and git runners и немного правок в апи

likvidator3301 opened the pull request:
new(api): logging in project controller
new(api): getAllProjects
new(api): camelCase in json serializer
new(utils): shell runner
new(utils): git runner with 'clone', 'checkout' and "getCommitHash'
new(repositories): dbSet in abstract repository's ctor
new(repositories): getAll

likvidator3301 committed:
new(api): logging in project controller

new(api): getAllProjects
new(api): camelCase in json serializer
new(utils): shell runner
new(utils): git runner with 'clone', 'checkout' and "getCommitHash'
new(repositories): dbSet in abstract repository's ctor
new(repositories): getAll

Mrtinic489 commented on Elevator.Api/Controllers/ProjectController.cs:
```diff
@@ -1,30 +1,44 @@
-        public async Task<OperationResult<CreateProjectResultDto>> Create([FromBody] CreateProjectRequestDto createProjectRequestDto)
+        public async Task<OperationResult<CreateProjectResultDto>> CreateAsync([FromBody] CreateProjectRequestDto createProjectRequestDto)
         {
+            logger.LogInformation($"Start execution method '{nameof(CreateAsync)}'");
```
А может добавим слово project в названия методов?
Мы как бы понимаем что раз они в контроллере проектов то это про них, но читается и ориентируется с ними получше. К тому же если мы сделаем аналогичный метод в другом контроллере, то как потом в логах отличать на какой из них пришёл запрос?

likvidator3301 replied:
У нас `ILogger<ProjectController>`
Он будет автоматически добавлять в начале сообщения свой скоуп - ProjectController
![image](https://user-images.githubusercontent.com/23081840/116215223-e8d67580-a760-11eb-8a1c-9d8d41fe4705.png)

Mrtinic489 commented on Elevator.Api/Startup.cs:
```diff
@@ -45,6 +53,8 @@ public void ConfigureServices(IServiceCollection services)
 
             services.AddScoped<IProjectService, ProjectService>();
 
+            services.AddLogging();
```
Ля, ето шо майкрософтовский логгер?)
Ничего против не имею, просто интересно))))

likvidator3301 replied:
yep

Mrtinic489 commented on Git/GitProject.cs:
```diff
@@ -0,0 +1,68 @@
+        private readonly ILoggerFactory loggerFactory;
+
+        public GitProject(GitProjectInformation gitProjectInformation, ILogger<GitProject> logger, ILoggerFactory loggerFactory)
+        {
```
IDevOps...

likvidator3301 replied:
да ну тут не получилось красивше

Mrtinic489 commented on Git/GitProject.cs:
```diff
@@ -0,0 +1,68 @@
+            {
+                var exitCode = await shellRunner.RunAsync(cancellationToken);
+                if (exitCode != 0)
+                    throw new CloneException(await shellRunner.ErrorStream.ReadToEndAsync());
```
Эмм, сами бросили, сами перехватили, и ещё раз бросили ошибку?
Можно по-подробней зачем?

Mrtinic489 commented on ShellRunner/ShellRunner.cs:
```diff
@@ -0,0 +1,82 @@
+using Shell.Exceptions;
+
+namespace Shell
+{
```
А мы не хотим вместо кидания ошибок в shellRunner заюзать аналог OperationResult?
Очень не хочется плодить try/catch в куче мест далее, т.к. shellRunner везде будет использоваться. 
Думаю стоит голосом обсудить это

likvidator3301 committed:
review fixes

new project - Common
operation result -> Common
git and shell runners now don't throw any exceptions
shell runner now return shell process

likvidator3301 committed:
hotfix

im colorblind i dont know
